arm-containerregistry/2016-06-27-preview/swagger/containerregistry.json

```diff
@@ -0,0 +1,642 @@
+      "description": "The storage account base properties.",
+      "properties": {
+        "name": {
+          "description": "Storage account name.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "RegistryCreateParameters": {
+      "description": "The parameters for creating a container registry.",
+      "required": [
+        "location",
+        "properties"
+      ],
+      "properties": {
+        "location": {
+          "description": "Resource location.",
+          "type": "string"
+        },
+        "tags": {
+          "description": "Resource tags.",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "properties": {
+          "$ref": "#/definitions/RegistryPropertiesCreateParameters",
+          "description": "The parameters for creating container registry properties.",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "RegistryPropertiesCreateParameters": {
+      "description": "The parameters for creating container registry properties.",
+      "required": [
+        "tenantId",
+        "storageAccount"
+      ],
+      "properties": {
+        "tenantId": {
+          "description": "The tenant id for the current user.",
+          "type": "string"
+        },
+        "adminUserEnabled": {
+          "description": "The boolean value that indicates whether admin user is enabled.",
+          "type": "boolean"
+        },
+        "storageAccount": {
+          "$ref": "#/definitions/StorageAccountProperties",
+          "description": "The storage account properties."
+        }
+      }
+    },
+    "StorageAccountProperties": {
+      "description": "The storage account properties.",
+      "required": [
+        "name",
+        "accessKey",
+        "endPointUrl"
+      ],
+      "properties": {
+        "name": {
+          "description": "Storage account name.",
+          "type": "string"
+        },
+        "accessKey": {
+          "description": "Storage account access key.",
+          "type": "string"
+        },
+        "endPointUrl": {
+          "description": "Storage account endpoint URL.",
+          "type": "string"
+        },
+        "location": {
+          "description": "Storage account location.",
+          "type": "string"
+        }
+      }
+    },
+    "Resource": {
+      "description": "Azure resource.",
+      "properties": {
+        "id": {
+          "description": "Resource Id.",
+          "type": "string",
+          "readOnly": true
+        },
+        "name": {
+          "description": "Resource name.",
+          "type": "string",
+          "readOnly": true
+        },
+        "type": {
+          "description": "Resource type.",
+          "type": "string",
+          "readOnly": true
+        },
+        "location": {
+          "description": "Resource location.",
+          "type": "string",
+          "readOnly": true

```

Comment:
To give you an idea, this is the model I was envisioning for StorageManagement spec and I believe Container Registry will have a very similar model. 

- **CreateOrUpdate {PUT}** (body parameter StorageAccount) -> StorageAccount
- **Update {PATCH}** (body parameter MutableStorageAccount) -> StorageAccount
- **Get {GET}** () -> StorageAccount

![image](https://cloud.githubusercontent.com/assets/4521223/19531202/a80b74c8-95eb-11e6-9373-65be43abbb55.png)

![image](https://cloud.githubusercontent.com/assets/4521223/19530891/93b3fab4-95ea-11e6-9eb3-4d8498947989.png)


Here, visualize 
- `StorageAccount` as `ContainerRegistry`
    - `StorageAccountProperties` as `ContainerRegistryProperties`
- `MutableStorageAccount` as `ContainerRegistryUpdateParameters`
     - `MutableStorageAccountProperties` as `ContainerRegistryPropertiesUpdateParameters`


- **CreateOrUpdate {PUT}** (body parameter ConatinerRegistry) -> ConatinerRegistry
- **Update {PATCH}** (body parameter ConatinerRegistryUpdateParameters) -> ConatinerRegistry
- **Get {GET}** () -> ConatinerRegistry

What do you think about it?